Makefile

```diff
@@ -243,44 +243,27 @@ $(BUILT)/add-on/install.rdf: add-on/install.rdf.in \
 # optimize this away.  We used the shared_jsx_temporary to make this code more
 # readable.
 shared_jsx_files=$(wildcard shared/js/*.jsx)
-built_ui_shared_js_files=$(shared_jsx_files:%.jsx=$(BUILT)/ui/%.js)
-built_add_on_shared_js_files=$(shared_jsx_files:%.jsx=$(BUILT)/add-on/chrome/content/%.js)
-built_standalone_shared_js_files=$(shared_jsx_files:%.jsx=$(BUILT)/standalone/content/%.js)
-
-# We can't use $(shared_jsx_files) here because % rules don't accept that.
-$(BUILT)/standalone/content/shared/js/%.js: shared/js/%.jsx
-	@mkdir -p $(@D)
-	$(BABEL) $< --out-file $@
-
-# We can't use $(shared_jsx_files) here because % rules don't accept that.
-$(BUILT)/add-on/chrome/content/shared/js/%.js: shared/js/%.jsx
-	@mkdir -p $(@D)
-	$(BABEL) $< --out-file $@
-
-# We can't use $(shared_jsx_files) here because % rules don't accept that.
-$(BUILT)/ui/shared/js/%.js: shared/js/%.jsx
-	@mkdir -p $(@D)
-	$(BABEL) $< --out-file $@
+shared_js_files=$(wildcard shared/js/*.js)
+shared_test_files=$(wildcard shared/test/*.js)
+built_ui_shared_js_files=$(shared_jsx_files:%.jsx=$(BUILT)/ui/%.js) $(shared_js_files:%.js=$(BUILT)/ui/%.js) $(shared_test_files:%.js=$(BUILT)/ui/%.js)
+built_add_on_shared_js_files=$(shared_jsx_files:%.jsx=$(BUILT)/add-on/chrome/content/%.js) $(shared_js_files:%.js=$(BUILT)/add-on/chrome/content/%.js) $(shared_test_files:%.js=$(BUILT)/add-on/chrome/content/%.js)
+built_standalone_shared_js_files=$(shared_jsx_files:%.jsx=$(BUILT)/standalone/content/%.js) $(shared_js_files:%.js=$(BUILT)/standalone/content/%.js) $(shared_test_files:%.js=$(BUILT)/standalone/content/%.js)
 
 #
 # The following sections are for the non-shared assets:
 #
 
 # ui-showcase
 ui_jsx_files=$(wildcard ui/*.jsx)
-built_ui_js_files=$(ui_jsx_files:%.jsx=$(BUILT)/%.js)
-
-$(BUILT)/ui/%.js: ui/%.jsx
-	@mkdir -p $(@D)
-	$(BABEL) $< --out-file $@
+ui_js_files=$(wildcard ui/*.js)
+built_ui_js_files=$(ui_jsx_files:%.jsx=$(BUILT)/%.js) $(ui_js_files:%.js=$(BUILT)/%.js)
```

Comment:
There's two places that are prefixing `$(BUILT)/` that would make it clear it's doing the same thing for all the files. Whereas with it right now, one needs to parse and recognize that each of the slightly different replacements are similar in some aspects but different in others.